v2/pkg/operators/matchers/validate.go

```diff
@@ -0,0 +1,51 @@
+package matchers
+
+import (
+	"fmt"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+// Validate perform initial validation on the matcher structure
+func (matcher *Matcher) Validate() error {
+	// uses yaml marshaling to convert the struct to map[string]interface to have same field names
+	matcherMap := make(map[string]interface{})
+	marshaledMatcher, err := yaml.Marshal(matcher)
+	if err != nil {
+		return err
+	}
+	if err := yaml.Unmarshal(marshaledMatcher, &matcherMap); err != nil {
+		return err
+	}
+
+	var unexpectedFields []string
+	switch matcher.matcherType {
+	case DSLMatcher:
+		unexpectedFields = []string{"status", "regex", "words", "size", "binary", "part"}
```

Comment:
instead of unexpected fields, wouldn't it be better to have expectedfields for each matcher type. That way we won't have repeated code.